accounts-db/src/accounts_db.rs

```diff
@@ -6758,20 +6758,22 @@ impl AccountsDb {
         txn_iter
             .enumerate()
             .map(|(i, txn)| {
-                let account = accounts_and_meta_to_store
-                    .account_default_if_zero_lamport(i)
-                    .map(|account| account.to_account_shared_data())
-                    .unwrap_or_default();
-                let account_info = AccountInfo::new(StorageLocation::Cached, account.lamports());
-
                 self.notify_account_at_accounts_update(
                     slot,
-                    &account,
+                    &accounts_and_meta_to_store
```

Comment:
This change will changes the semantic of account update. I think the problem can be solved better by either an explicit new notification (account delete), or update the account update interface with the previous pubkey info.